### PR TITLE
Embed public Beam IO Performance Metrics

### LINKED ISSUE #27539: [Task]: Embed public Beam IO Performance Metrics
### What needs to happen?

The output of this task are embedded visualizations of Beam IO performance metrics.

### Issue Priority

Priority: 2 (default / most normal work should be filed as P2)

### Issue Components

- [ ] Component: Python SDK
- [ ] Component: Java SDK
- [ ] Component: Go SDK
- [ ] Component: Typescript SDK
- [ ] Component: IO connector
- [ ] Component: Beam examples
- [ ] Component: Beam playground
- [ ] Component: Beam katas
- [X] Component: Website
- [ ] Component: Spark Runner
- [ ] Component: Flink Runner
- [ ] Component: Samza Runner
- [ ] Component: Twister2 Runner
- [ ] Component: Hazelcast Jet Runner
- [ ] Component: Google Cloud Dataflow Runner

### PR DESCRIPTION
**Pull request by damondouglas**
This PR closes #27539 by embedding Looker visualizations of Beam IO performance metrics.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - ~Update `CHANGES.md` with noteworthy changes.~
 - ~If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).~

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

File: website/www/site/data/performance.yaml [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+host: https://28cd0c7f-7a84-428c-bfa7-6e8316f6931b.looker.app
+path: looks
+looks:
+  bigquery:
+    read:
+      test_name:
+        - id: nwQxvsnQFdBPTk27pZYxjcGNm2rRfNJk
```
Comment (Abacn) [inline website/www/site/data/performance.yaml:19]: it appears only a few graphs render successfully, e.g. http://apache-beam-website-pull-requests.storage.googleapis.com/27540/performance/bigquery/index.html

may just show "Runtime and Estimated Cost", "AvgIn/OutputThroughputBytesPerSec", "AvgIn/OutputThroughputElementsPerSec" plots that's all we need

File: website/www/site/layouts/partials/header.html [hunk 2]
```diff
@@ -81,6 +81,15 @@
        <li>
          <a class="navbar-link" href={{ "/case-studies/" | relLangURL }}>{{ T "nav-case-studies" }}</a>
        </li>
+        <li>
+          <span class="navbar-link">Performance</span>
```
Comment (Abacn) [inline website/www/site/layouts/partials/header.html:85]: we may put the dashboards under documentation/io/connectors at this moment; we may or may not need this as an extra top level tab on Tab when this is polished

**Review by Abacn — COMMENTED** [review]
Since this is in draft, just added a few comments for the possible simplified version ready to show publicly

File: website/www/site/data/performance.yaml (hunk 1 above)
Comment (Abacn) [reply website/www/site/data/performance.yaml:20]: From staging http://apache-beam-website-pull-requests.storage.googleapis.com/27540/performance/bigtable/index.html now there is still 20+ plots shown. Please just show "Runtime and Estimated Cost", "AvgIn/OutputThroughputBytesPerSec", "AvgIn/OutputThroughputElementsPerSec" plots that's all we need (for batch)

File: website/www/site/layouts/partials/header.html (hunk 2 above)
Comment (Abacn) [reply website/www/site/layouts/partials/header.html:85]: what do you think @pranavbhandari24 @liferoad

Comment (liferoad) [reply website/www/site/layouts/partials/header.html:85]: @damondouglas could we move this to `documentation/io/connectors`? I do think later we can move this back to the top level once they are polished.

File: website/www/site/data/performance.yaml (hunk 1 above)
Comment (liferoad) [reply website/www/site/data/performance.yaml:20]: @damondouglas can we just show what @Abacn list for now?

File: website/www/site/layouts/partials/header.html (hunk 2 above)
Comment (Abacn) [reply website/www/site/layouts/partials/header.html:85]: If this is non-trivial we can just comment out the top level navbar at this moment (and add a entry link in io documentation)

File: website/www/site/data/performance.yaml (hunk 1 above)
Comment (damondouglas) [reply website/www/site/data/performance.yaml:20]: Per our discussion in chat I will resolve this.

File: website/www/site/layouts/partials/header.html (hunk 2 above)
Comment (damondouglas) [reply website/www/site/layouts/partials/header.html:85]: Per our discussion in chat I will resolve this.

File: website/www/site/content/en/performance/glossary/_index.md [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+The total Solid State Disk usage of the pipeline Job.
+
+## TotalStreamingDataProcessed
+
+The total amount of streaming data processed by the pipeline Job.
+
+## TotalVcpuTime
+
+The total amount of virtual CPU time used by the pipeline Job.
```
Comment (Abacn) [inline website/www/site/content/en/performance/glossary/_index.md:120]:
```suggestion
## AvgInputThroughputBytesPerSec

The mean input throughput of the pipeline Job measured in bytes per second.

## AvgInputThroughputElementsPerSec

The mean elements input throughput per second of the pipeline Job.

## AvgOutputThroughputBytesPerSec

The mean output throughput of the pipeline Job measured in bytes per second.

## AvgOutputThroughputElementsPerSec

The mean elements output throughput per second of the pipeline Job.

## EstimatedCost

The estimated cost of the pipeline Job.

## RunTime

The time it took for the pipeline Job to run measured in seconds.
```

File: website/www/site/data/performance.yaml [hunk 4]
```diff
@@ -0,0 +1,304 @@
...
+      - id: 9Yz9m7ymhch396q3ksYbgSBTn5FDdCyf
+        title: RunTime by Version
+      - id: tsPk58rZGsHzfDXCSXtTbkSQXH75tNqs
+        title: EstimatedDataProcessedGB by Version
+      - id: 2YG6R9Z9pZGy4jPBDMXsGBbkxvWkpx66
+        title: MaxCpuUtilization by Version
+      - id: c5s8jGZFfvzPTHcYHY57RvWmGVKw3fDw
+        title: AvgCpuUtilization by Version
+      - id: TnKzJxZXRqZQmD32cHF3y5XXnfzTKsqH
+        title: EstimatedCost by Version
```
Comment (Abacn) [inline website/www/site/data/performance.yaml:304]:
```suggestion
looks:
  bigquery:
    read:
      folder: 30
      test_name:
        - id: nwQxvsnQFdBPTk27pZYxjcGNm2rRfNJk
          title: Runtime and Estimated Cost by Test Name
      date:
        - id: 7QKbMmgT5NgPH6RsDfVQmKMdjJysS37x
          title: AvgOutputThroughputBytesPerSec by Date
        - id: hSqCyCkHkh4whNZSpdVFwB8ksPWrbScb
          title: AvgOutputThroughputElementsPerSec by Date
      version:
        - id: SghzxqjMb5QQwgZn7yHw4MnpGXBgXQft
          title: AvgOutputThroughputBytesPerSec by Version
        - id: QT6X4GTCnxxykSZ7tpXyBfBFCKMgdzs7
          title: AvgOutputThroughputElementsPerSec by Version
    write:
      folder: 31
      test_name:
        - id: sHyRfwCfGCPqSwsTWXSsxfbHKH5hXjzc
          title: Write BigQuery RunTime and EstimatedCost
      date:
        - id: p2qnxS58WdWdZTzJhXfSbkhxGyDXMYfY
          title: AvgInputThroughputBytesPerSec by Date
        - id: Bmg6sv8XPtcn3cGtgPtKP26Zj4cyd2XK
          title: AvgInputThroughputElementsPerSec by Date
      version:
        - id: Q6QvktyfSFrSPq4TSDgQQyV9jttjwQJp
          title: AvgInputThroughputBytesPerSec by Version
        - id: ktfyZ7Th8yRFFwWgb4WSxngBpmSfz8xh
          title: AvgInputThroughputElementsPerSec by Version
  bigtable:
    read:
      folder: 32
      test_name:
        - id: YQ2W3wdNnBXMgDgpzCRbmQWMHjyPvZny
          title: Read BigTable RunTime and EstimatedCost
      date:
        - id: szvwNfPrwTtmRmMHWv3QFh6wTKxm26TF
          title: AvgOutputThroughputBytesPerSec by Date
        - id: ZRQw7np2mj35kQHcvgtNDwVGgP855QNF
          title: AvgOutputThroughputElementsPerSec by Date
      version:
        - id: X7y2hQQkJnYY8ctnr2y7NnNx7xrMkjt2
          title: AvgOutputThroughputBytesPerSec by Version
        - id: B3HwmTtn9nBfzwRC7s3WfQSjf7ckrfXx
          title: AvgOutputThroughputElementsPerSec by Version
    write:
      folder: 33
      test_name:
        - id: 2sC27RQwWy2MP9DXVHjbvTYSNFYpFvxj
          title: Write BigTable RunTime and EstimatedCost
      date:
      - id: X22sDqD8krBQQ4mRXTFMmpKFvgkZ529g
        title: AvgInputThroughputBytesPerSec by Date
      - id: WqHVhrjQVrrjpczqgJB9s9k8fvt3Vc2d
        title: AvgInputThroughputElementsPerSec by Date
      version:
      - id: j72shCBz6rJhQP8JwYB3JcVrhv9BWGpD
        title: AvgInputThroughputBytesPerSec by Version
      - id: 6DyhkfvcNfkJg53hwFnRzxqbJDdGb5t7
        title: AvgInputThroughputElementsPerSec by Version
  textio:
    read:
      folder: 34
      test_name:
        - id: SsfFgvwyMthnHjdRwBjWGNbvfNgym4wb
          title: Read TextIO RunTime and EstimatedCost
      date:
        - id: Pr8MBG66JVgBmdQbDr6HXtRprjr3ybj6
          title: AvgOutputThroughputBytesPerSec by Date
        - id: CTZYK4KVYM65Zjn6jWgDYYNBWckDYKRQ
          title: AvgOutputThroughputElementsPerSec by Date
      version:
        - id: Dcvfh3XFZySrsmPY4Rm8NYyMg5QQRBF6
          title: AvgOutputThroughputBytesPerSec by Version
        - id: dN8mTZsVZc7vGDYKJCT8S67BCXzVJT4s
          title: AvgOutputThroughputElementsPerSec by Version
    write:
      folder: 35
      test_name:
        - id: Mm2j2QPc2x4hZqYNSpZC4bDpH2fgvqvp
          title: Write TextIO RunTime and EstimatedCost
      date:
      - id: J9VXhp3ry5zbPFFGsYNfDRypGNVNMbPV
        title: AvgInputThroughputBytesPerSec by Date
      - id: KwSCnyz75wMpXhGpZQ8FRp6cwt8pjhfD
        title: AvgInputThroughputElementsPerSec by Date
      version:
      - id: VFXbPV9JGJxmNYnGsypQzH97RPDFjpPN
        title: AvgInputThroughputBytesPerSec by Version
      - id: fVVHhXCrHNgBG52TJsTjR8VbmWCCQnVN
        title: AvgInputThroughputElementsPerSec by Version
```

**Review by Abacn — APPROVED** [review]
LGTM, thanks!

**Pull request merged**